CMakeLists.txt

```diff
@@ -15,8 +15,10 @@ set(CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
+if (NOT BUILD_SHARED_LIBS)
+  set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+  set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
+endif ()
```

Comment:
@compnerd Is there a "right" thing to do with respect to symbol visibility in shared libs that doesn't involve a ton of work annotating everything in our APIs?

Reply:
@bertmaher, no, annotations are the right thing to do unfortunately.